app/bundles/ReportBundle/Views/Report/details.html.twig

```diff
@@ -0,0 +1,158 @@
+{% set header = 'mautic.report.report.header.view'|trans({'%name%': report.name|trans|escape}) %}
+{% set showDynamicFilters = report.settings.showDynamicFilters is defined %}
+{% set hideDateRangeFilter = report.settings.hideDateRangeFilter is defined %}
+
+{% block mauticContent %}report{% endblock %}
+{% block headerTitle %}{{ header }}{% endblock %}
+
+{% if 'index' == tmpl %}
+    {% extends('MauticCoreBundle:Default:content.html.twig') %}
+
+    {% set buttons = [] %}
+    {% if data is not empty or graphs is not empty %}
+        {% set buttons = buttons|merge([{
+            'attr'      : {
+                'target'        : '_new',
+                'data-toggle'   : '',
+                'class'         : 'btn btn-default btn-nospin',
+                'href'          : path('mautic_report_export', {'objectId':report.id, 'format':'html'})
+            },
+            'btnText'   : 'mautic.form.result.export.html'|trans,
+            'iconClass' : 'fa fa-file-code-o'
+        }]) %}
+        {% if data is not empty %}
+            {% set buttons = buttons|merge([{
+                'attr'      : {
+                    'data-toggle'   : 'download',
+                    'class'         : 'btn btn-default btn-nospin',
+                    'href'          : path('mautic_report_export',
+                        {'objectId':report.id, 'format':'csv'})
+                },
+                'btnText'   : 'mautic.form.result.export.csv'|trans,
+                'iconClass' : 'fa fa-file-text-o'
+            }]) %}
+            {% if spreadsheetExist %}
```

Comment:
I have a branch that includes some new tests to check for this kind of thing. Reference https://github.com/mautic/mautic/pull/11914/files#diff-c9803ecc10c8fb93dfcb9501f0913395df3ef9172e7fad23e6e43b366b477f7e (`app/bundles/CoreBundle/Templating/Twig/Extension/AppExtension.php` file)

You can pull files in from other branches with the command `git checkout {branch} {file}` or just copy and we can resolve the conflict later.